internal/monitors/repotester/repotester.go

```diff
@@ -0,0 +1,86 @@
+package repotester
+
+import (
+	"context"
+
+	argoapi "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/v2/util/git"
+	api "github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/service"
+	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/util"
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/flightctl/flightctl/pkg/reqid"
+	"github.com/go-chi/chi/v5/middleware"
+	"github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+)
+
+type API interface {
+	Test()
+}
+
+type RepoTester struct {
+	log       logrus.FieldLogger
+	db        *gorm.DB
+	repoStore service.RepositoryStoreInterface
+}
+
+func NewRepoTester(log logrus.FieldLogger, db *gorm.DB, store *store.Store) *RepoTester {
+	return &RepoTester{
+		log:       log,
+		db:        db,
+		repoStore: store.GetRepositoryStore(),
+	}
+}
+
+func (r *RepoTester) Test() {
+	reqid.OverridePrefix("repotester")
```

Comment:
Are you sure that it depend on the package that import the `reqid`? wouldn't that affect all the loggers? 
We can change the basic request id to be something else if needed and hostname is not required. to log specific package you can add another field to the logger like `r.log = r.log.WithField("pkg", "repotester")`

Reply:
I tested this by adding another thread.  The logs showed each thread with its own prefix and reqid.